rdr_service/resource/generators/participant.py

```diff
@@ -1062,16 +1015,45 @@ def get_biobank_sample_activity(samples):
                     orderless_stored_samples.append(sr)
 
             order = {
+                'finalized_time': None,
                 'biobank_order_id': 'UNSET',
                 'tests_stored': len(orderless_stored_samples),
-                'samples': orderless_stored_samples
+                'samples': orderless_stored_samples,
+                'isolate_dna': sum([r['dna_test'] for r in orderless_stored_samples]),
+                'isolate_dna_confirmed': 0,  # Fill in below.
+                'baseline_test': sum([r['baseline_test'] for r in orderless_stored_samples]),
+                'baseline_tests_confirmed': 0  # Fill in below.
             }
             orders.append(order)
-            if order['samples']:
-                activity = activity + get_biobank_sample_activity(order['samples'])
 
         if len(orders) > 0:
             data['biobank_orders'] = orders
+
+            # Calculate confirmed tests and save activity events.
+            for order in orders:
+                act_key = ParticipantEventEnum.BiobankOrder
+                act_ts = order['finalized_time']
+                if act_ts:
+                    act_key = ParticipantEventEnum.BiobankShipped
+
+                if order['samples']:
+                    order['baseline_tests_confirmed'] = \
+                        sum([r['baseline_test'] for r in order['samples'] if r['confirmed'] is not None])
+                    order['isolate_dna_confirmed'] = \
+                        sum([r['dna_test'] for r in order['samples'] if r['confirmed'] is not None])
+                    # Find minimum confirmed date of DNA tests if we have any.
+                    if order['isolate_dna']:
+                        try:
+                            tmp_ts = min([r['confirmed'] for r in order['samples'] if r['confirmed'] is not None])
```

Comment:
I think it's pretty typical for all samples in an order to share timestamps like confirmed...but if a single biobank order can share both DNA and non-DNA sample types,  should this be:

min([r['confirmed'] for r in order['samples'] if ( r['confirmed'] is not None and r['dna_test']))